tests/unit/test_saml.py

```diff
@@ -155,3 +156,57 @@ def test_saml_with_valid_metadata(mock_urlopen, metadata_file, binding):
         assert endpoints[1].binding == binding
         assert endpoints[1].url == "https://login.staging.ubuntu.com/saml/"
         assert endpoints[1].response_url is None
+
+
+@patch.object(ssl, "get_server_certificate", return_value="somecert")
+def test_saml_with_invalid_certificate(mock_get_server_certificate):
+    """
+    arrange: mock certificate retrieved from the metadata URL.
+    act: validate the certificate.
+    assert: the certificate is invalid.
+    """
+    harness = Harness(SamlIntegratorOperatorCharm)
+    harness.begin()
+    harness.disable_hooks()
+    entity_id = "https://login.staging.ubuntu.com"
+    metadata_url = "https://login.staging.ubuntu.com/saml/metadata"
+    harness.update_config(
+        {
+            "entity_id": entity_id,
+            "metadata_url": metadata_url,
+            "certificate": "someothercert",
+        }
+    )
+    charm_state = CharmState.from_charm(harness.charm)
+    try:
+        SamlIntegrator(charm_state=charm_state)
```

Comment:
Couldnt  "with pytest.raises" be used here?

Reply:
Thx for the feeedback!